### ISSUE TITLE
Tuning parameters for logo dataset

### ISSUE THREAD

**Issue by erdavis1** [lucidrains/unet-stylegan2#6]
Hi! First off, thank you for being so engaged and active with folks' issues.

I'm training on a dataset of 3k Japanese flag logos, plus their rotations of 90, 180, and 270 degrees for 12k total images. 

I generally experience very quick mode collapse, often within the first 5k iterations. The one result that it does produce is pretty nice, though!
`unet_stylegan2  --data jpegs/ --attn-layers [1,2] `
39k iterations:
![image](https://user-images.githubusercontent.com/45726527/97325323-13865680-1830-11eb-9816-d11adb6b4768.png)

Increasing the batch size and gradient accumulate every has reduced it to partial mode collapse. The catch is I'm training on colab, and this can be unworkably slow depending on the GPU (30 secs per iteration)
`unet_stylegan2  --data jpegs/  --batch-size 32 --gradient-accumulate-every 8 --cl-reg --attn-layers [1,2] `

24k iterations
![image](https://user-images.githubusercontent.com/45726527/97326203-fe5df780-1830-11eb-9036-876e476f370f.png)

Do you have any tuning suggestions for increasing speed and avoiding mode collapse? The results at 24k (above) are promising, but definitely need a lot more refining!

**Comment by lucidrains** [comment]
@erdavis1 you are working in the realm of small data, so i would suggest turning on augmentation! you can do so with `--aug-prob 0.25`

**Comment by lucidrains** [comment]
@erdavis1 there actually isn't any contrastive regularization in unet-stylegan2, i removed it since it doesn't fit within the unet discriminator framework

**Comment by erdavis1** [comment]
@lucidrains Thanks! I'm very new to this, if it isn't obvious.

I tried the following and also got pretty quick mode collapse.

`!unet_stylegan2  --data jpegs/ --attn-layers [1,2] --aug-prob 0.25`
9k iterations:
![image](https://user-images.githubusercontent.com/45726527/97378657-17da6000-1880-11eb-9358-4f67106b987c.png)

I feel like drastically increasing the batch size has given me the best results of what I've tried, but I'm really struggling to find a balance where my results are good+diverse, but it doesn't take >15secs per iteration to train.

Are there any other tricks to consider, aside from batch size?

**Comment by lucidrains** [comment]
Have you tried plain stylegan2, without the unet? github.com/lucidrains/stylegan2-pytorch

**Comment by erdavis1** [comment]
I have. with very similar results

**Comment by lucidrains** [comment]
@erdavis1 try with augmentation, but with an effective batch size of 64. That means just a batch size of 32 and the gradient accumulate every of 2

**Comment by erdavis1** [comment]
Thanks! I'll try running that today and report back.

**Comment by erdavis1** [comment]
The following ended up in a very quick mode collapse, unfortunately:
`unet_stylegan2  --data jpegs/ --attn-layers [1,2]  --batch-size 32 --gradient-accumulate-every 2 --aug-prob 0.25`

1k:
![image](https://user-images.githubusercontent.com/45726527/97578690-638b2780-19ae-11eb-9847-200f71bc51f3.png)

20k:
![image](https://user-images.githubusercontent.com/45726527/97578710-68e87200-19ae-11eb-9ba9-0066db08ee11.png)

For my knowledge, what differences would you expect to see in a model trained with an effective batch size of 64 (batch size 32, gradient accumulate every 2) vs. one also with effective batch size 64 (batch size 8, gradient accumulate every 8)

**Comment by lucidrains** [comment]
@erdavis1 i updated the augmentation code in v0.5.0 if you'd like to give that a try. `--aug-prob 0.25 --aug-types [translation,cutout,color]`

**Comment by lucidrains** [comment]
@erdavis1 get more data! lol

**Comment by erdavis1** [comment]
Haha, if only I could!

I ended up being pretty happy with the results of this run, even though it was fairly slow (~7 sec/iter if I got a V100)

`!unet_stylegan2  --data jpegs/  --attn-layers [1,2]  --batch-size 32 --gradient-accumulate-every 8 --aug-prob 0.25   --aug-types [translation,cutout,color]`

Closing the issue since I'm satisfied with the output.

17k:
![image](https://user-images.githubusercontent.com/45726527/98745876-402a8a00-2369-11eb-8ac4-e137c1f8e684.png)

**Comment by lucidrains** [comment]
🚀🚀
